docs/MigrationGuide.md

```diff
@@ -0,0 +1,118 @@
+# Migration Guide from unleash-android-proxy-sdk
+
+This guide provides detailed steps for migrating your project from the Unleash Android Proxy SDK version to the newer Unleash Android SDK.
+
+We will focus on the [previous sample application](https://github.com/Unleash/unleash-android-proxy-sdk/tree/main/samples/android) and specifically highlighting the changes from this pull request: https://github.com/Unleash/unleash-android-proxy-sdk/pull/83
+
+## Benefits of Migrating
+
+This version of the Unleash Android SDK introduces several improvements, including:
+- No need for SLF4J dependency and the usage of the native Android logging system.
+- Respecting the Android lifecycle to stop polling and sending metrics in the background.
+- The new SDK respects the minimum Android API level 21, but we recommend API level 23.
+- Monitoring network connectivity to avoid unnecessary polling (requires API level 23 or above).
+
+## Overview
+
+The new SDK introduces several changes and improvements, including API modifications and a new initialization process.
+
+## Step-by-Step Migration
+
+### 1. Update Gradle Dependency
+
+First, update the dependency in your `build.gradle` file:
+
+```gradle
+dependencies {
+    // Remove the old SDK
+    implementation 'io.getunleash:unleash-android-proxy-sdk:0.5.0'
+
+    // Add the new SDK
+    implementation 'io.getunleash:unleash-android:$version'
+}
+```
+
+### 2. Update the initialization code
+We won't cover all the details here as most of the configuration can be set using the builders fluent methods. However, the main difference is that the new SDK requires an `Application` context to be passed to the `Unleash` constructor. This is necessary to monitor the network connectivity and respect the Android lifecycle (usually it can be injected if using hilt with `@ApplicationContext context`).
```

Comment:
That last parenthesis is a little jumbled, but I lack a lot of context. What is hilt? Some DI system? How about unpacking it into its own sentence and giving it some more room to breathe? Something like this
```suggestion
We won't cover all the details here as most of the configuration can be set using the builders fluent methods. However, the main difference is that the new SDK requires an `Application` context to be passed to the `Unleash` constructor. This is necessary to monitor the network connectivity and respect the Android lifecycle. If you use hilt, this can be injected with `@ApplicationContext context`.
```